src/main/kotlin/ecommerce/service/ProductService.kt

```diff
@@ -4,22 +4,39 @@ import ecommerce.dto.ProductForm
 import ecommerce.exception.InternalServerErrorException
 import ecommerce.exception.NotFoundException
 import ecommerce.exception.ProductNameAlreadyExistsException
+import ecommerce.model.Option
 import ecommerce.model.Product
+import ecommerce.repository.OptionRepository
 import ecommerce.repository.ProductRepository
+import org.springframework.data.domain.Page
+import org.springframework.data.domain.PageRequest
+import org.springframework.data.domain.Sort
 import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
 
 @Service
-class ProductService(private val productRepository: ProductRepository) {
+class ProductService(
+    private val productRepository: ProductRepository,
+    private val optionRepository: OptionRepository,
+) {
     fun insert(form: ProductForm): Product {
-        checkProductNameExists(form.name)
-        val product = ProductForm.toProduct(form)
+        nameExists(form.name)
+        val option = Option(name = "none", quantity = 1)
+        val product = ProductForm.toProduct(form, listOf(option))
+        option.product = product
```

Comment:
I don't think this in unnecessary because when you make `Product`, it is setting itself to the option in the constructor (`init` block)